lib/realtime/tenants/authorization.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Realtime.Tenants.Authorization do
+  alias Plug.Conn
+  alias Phoenix.Socket
+
+  require Logger
+
+  def get_authorizations(%Socket{} = socket, db_conn, params) do
+    with {:ok, {:ok, permissions}} <- get_permissions_for_connection(db_conn, params) do
+      {:ok, Socket.assign(socket, :permissions, permissions)}
+    else
+      {:error, _} -> {:error, :unauthorized}
+    end
```

Comment:
nitpick and a small fix of the nested `:ok` tuple
```suggestion
    case get_permissions_for_connection(conn, params) do
      {:ok, permissions} -> {:ok, assign(socket, :permissions, permissions)}
      _ -> {:error, :unauthorized}
    end
```